plugin/tls/provider.go

```diff
@@ -0,0 +1,157 @@
+package tls
+
+import (
+	"os"
+
+	"github.com/caddyserver/certmagic"
+	"github.com/libdns/alidns"
+	"github.com/libdns/azure"
+	"github.com/libdns/cloudflare"
+	"github.com/libdns/digitalocean"
+	"github.com/libdns/duckdns"
+	"github.com/libdns/googleclouddns"
+	"github.com/libdns/hetzner"
+	"github.com/libdns/ionos"
+	"github.com/libdns/linode"
+	"github.com/libdns/namecheap"
+	"github.com/libdns/netlify"
+	"github.com/libdns/powerdns"
+	"github.com/libdns/route53"
+	"github.com/libdns/vercel"
+)
+
+func createDNSSolver(provider string) *certmagic.DNS01Solver {
+	var dnsSolver *certmagic.DNS01Solver
+	switch provider {
+	case "digitalocean":
```

Comment:
Okay I am starting to see why caddy doesn't pack all the providers - I haven't added all of them and the binary is already 13M bigger than before. 

Any thoughts on this @SuperQ @chrisohaver ?

Does it make sense to limit this to only the most popular ones (and if so who decides which are those)?

Reply:
Perhaps it could make sense to provide these as external plugins, one for each provider, this would keep the binary size small. The only downside I see with that is that it would be a lot of code duplication as these would be mostly identical. 

Users would then just have to pick the plugin for their dns provider

Reply:
Created one as an example: https://github.com/mariuskimmina/acme_digitalocean